test/fork/integration.test.js

```diff
@@ -404,4 +407,62 @@ describe('Babylon integrations', function () {
       });
     }
   }
+
+  it('can deploy a strategy with the Liquity Integration', async () => {
+
+    const customIntegration = await deploy('CustomIntegrationLiquity', {
+      from: alice.address,
+      args: [controller.address, LIQUITY_BORROWER_OPERATIONS_ADDDRESS],
+    });
+
+    await garden.connect(alice).addStrategy(
+      'Liquity',
+      '💎',
+      [
+        eth(10), // maxCapitalRequested: eth(10),
+        eth(0.1), // stake: eth(0.1),
+        ONE_DAY_IN_SECONDS * 30, // strategyDuration: ONE_DAY_IN_SECONDS * 30,
+        eth(0.05), // expectedReturn: eth(0.05),
+        eth(0.1), // maxAllocationPercentage: eth(0.1),
+        eth(0.05), // maxGasFeePercentage: eth(0.05),
+        eth(0.09), // maxTradeSlippagePercentage: eth(0.09),
+      ],
+      [5], // _opTypes
+      [customIntegration.address], // _opIntegrations
+      new ethers.utils.AbiCoder().encode(
+        ['address', 'uint256'],
+        [ADDRESS_ZERO, eth(3000)] // integration params, 3000 LUSD borrow
+      ), // _opEncodedDatas
+    );
+
+    const strategies = await garden.getStrategies();
+    customStrategy = await ethers.getContractAt('IStrategy', strategies[0]);
+
+    await garden.connect(alice).deposit(eth(10), 0, alice.address, ADDRESS_ZERO, {
+      value: eth(10),
+    });
+    const balance = await garden.balanceOf(alice.getAddress());
+
+    // Vote Strategy
+    await customStrategy.connect(keeper).resolveVoting([alice.address], [balance], 0);
+
+    // Execute strategy
+    await increaseTime(ONE_DAY_IN_SECONDS);
+    await customStrategy.connect(keeper).executeStrategy(eth(10), 0);
+
+    // Need additional LUSD to repay debt
+    await hre.network.provider.request({
```

Comment:
Needed some way to add additional LUSD to our strategy. I suppose I could chain some other kind of trade operation or something to the test? I wasn't sure how to do that, not sure if it's documented.

Reply:
Why do you need extra? If you don't spend the LUSD you request when opening the trove you can use it to repay the debt right? You can check the example where we chain a curve operation and enter a curve pool for example after this. Then on exit, we would sell the curve lp tokens and rewards and use them to pay the debt

Reply:
Here is my understanding of Liquity's flow:

<img width="728" alt="Screen Shot 2022-06-05 at 6 02 39 PM" src="https://user-images.githubusercontent.com/1270317/172078512-4f0d0be0-89be-4b75-878c-313ba9b0fd01.png">


We deposit 5 ETH and borrow 2,000 LUSD.

We actually incur a debt of 2,210 LUSD:

- 200 LUSD is the liquidation reserve, which is repaid to us when we repay our debt
- 10 LUSD is the borrowing fee, which is not repaid to us

Therefore, to repay our debt we need 2,010 LUSD. (At which point we will receive all 5 ether that we deposited.)

Reply:
Nice, thank you for the detailed explanation. It is clear now